lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/TableIndexDAOImplUnitTest.java

```diff
@@ -208,4 +212,97 @@ public void testValidateMaxListLengthInAnnotationReplication_allUnderLimit() {
 		verify(spyDao).getMaxListSizeForAnnotations(scopeFilter, listAnnotationNames, objectIdFilter);
 	}
 	
+	// relevant to PLFM-6344
+	@Test
+	public void testExpandFromAggregationWithZeroMaxStringElementSize() {
+		
+		// case 1: string column type, max string size is 0
+		ColumnAggregation zero = new ColumnAggregation();
+		zero.setColumnName("foo");
+		zero.setColumnTypeConcat(concatTypes(AnnotationType.STRING));
+		zero.setMaxStringElementSize(0L);
+		zero.setMaxListSize(1L);
+		
+		// case 2: string and double column type, max string size is 0
+		ColumnAggregation one = new ColumnAggregation();
+		one.setColumnName("bar");
+		one.setColumnTypeConcat(concatTypes(AnnotationType.STRING, AnnotationType.DOUBLE));
+		one.setMaxStringElementSize(0L);
+		one.setMaxListSize(1L);
+
+		// case 3: string list column type, max string size is 0
+		ColumnAggregation two = new ColumnAggregation();
+		two.setColumnName("foobar");
+		two.setColumnTypeConcat(concatTypes(AnnotationType.STRING));
+		two.setMaxStringElementSize(0L);
+		two.setMaxListSize(3L);
+		
+		// case 4: string and double list column type, max string size is 0
+		// 		   string and double column type, max string size is 0
+		ColumnAggregation three = new ColumnAggregation();
+		three.setColumnName("barbaz");
+		three.setColumnTypeConcat(concatTypes(AnnotationType.STRING, AnnotationType.DOUBLE));
+		three.setMaxStringElementSize(0L);
+		three.setMaxListSize(3L);
+
+		// call under test
+		List<ColumnModel> results = TableIndexDAOImpl.expandFromAggregation(Lists.newArrayList(zero,one,two,three));
+		assertEquals(6, results.size());
+		
+		// case 1: string column type with 0L max size, should give 50L max size
+		ColumnModel cm = results.get(0);
+		assertEquals("foo", cm.getName());
+		assertEquals(ColumnType.STRING, cm.getColumnType());
+		assertEquals(null, cm.getMaximumListLength());
+		assertEquals(50L, cm.getMaximumSize());
```

Comment:
Instead of 50L might be better to use the ColumnConstants.DEFAULT_STRING_SIZE value